django/contrib/admin/templatetags/admin_modify.py

```diff
@@ -43,14 +44,13 @@ def submit_row(context):
     """
     Displays the row of buttons for delete and save.
     """
-    opts = context['opts']
     change = context['change']
     is_popup = context['is_popup']
     save_as = context['save_as']
     show_save = context.get('show_save', True)
     show_save_and_continue = context.get('show_save_and_continue', True)
-    ctx = {
-        'opts': opts,
+    ctx = Context(context)
```

Comment:
Editing the existing context (which is instance of `RequestContext` causes 3 failures in `admin_views.tests.SaveAsTests`, probably because the same dict is modified in different stages of execution.